db/compaction/compaction_service_job.cc

```diff
@@ -475,6 +477,195 @@ static std::unordered_map<std::string, OptionTypeInfo> cs_input_type_info = {
+          OptionType::kEncodedString}},
+        {"filter_policy_name",
+         {offsetof(struct TableProperties, filter_policy_name),
+          OptionType::kEncodedString}},
+        {"comparator_name",
+         {offsetof(struct TableProperties, comparator_name),
+          OptionType::kEncodedString}},
+        {"merge_operator_name",
+         {offsetof(struct TableProperties, merge_operator_name),
+          OptionType::kEncodedString}},
+        {"prefix_extractor_name",
+         {offsetof(struct TableProperties, prefix_extractor_name),
+          OptionType::kEncodedString}},
+        {"property_collectors_names",
+         {offsetof(struct TableProperties, property_collectors_names),
+          OptionType::kEncodedString}},
+        {"compression_name",
+         {offsetof(struct TableProperties, compression_name),
+          OptionType::kEncodedString}},
+        {"compression_options",
+         {offsetof(struct TableProperties, compression_options),
+          OptionType::kEncodedString}},
+        {"seqno_to_time_mapping",
+         {offsetof(struct TableProperties, seqno_to_time_mapping),
+          OptionType::kEncodedString}},
+        {"user_collected_properties",
+         {offsetof(struct TableProperties, user_collected_properties),
+          OptionType::kVector, OptionVerificationType::kByNameAllowNull,
+          OptionTypeFlags::kCompareNever,
+          [](const ConfigOptions& /*opts*/, const std::string& /*name*/,
+             const std::string& value, void* addr) {
+            UserCollectedProperties properties;
+            Status s;
+            for (size_t start = 0, end = 0;
+                 s.ok() && start < value.size() && end != std::string::npos;
+                 start = end + 1) {
+              std::string token;
+              s = OptionTypeInfo::NextToken(value, ';', start, &end, &token);
+              if (s.ok() && !token.empty()) {
+                std::vector<std::string> kvPair = StringSplit(token, '=');
+                assert(kvPair.size() == 2);
+                std::string decoded_key;
+                std::string decoded_value;
+                (Slice(kvPair[0])).DecodeHex(&decoded_key);
+                (Slice(kvPair[1])).DecodeHex(&decoded_value);
+                properties.emplace(std::move(decoded_key),
+                                   std::move(decoded_value));
+              }
+            }
+            if (s.ok()) {  // It worked
+              *(static_cast<UserCollectedProperties*>(addr)) = properties;
+            }
+            return s;
+          },
+          [](const ConfigOptions& /*opts*/, const std::string& /*name*/,
+             const void* addr, std::string* value) {
+            const auto properties =
+                static_cast<const UserCollectedProperties*>(addr);
+            value->append("{");
+            for (const auto& prop : *properties) {
+              value->append(Slice(prop.first).ToString(true));
```

Comment:
Just copied how `kEncodedString` type is serialized in options_helper. I think I can probably introduce a new OptionsType, `OptionsType::StringMap` or something, but I wanted to get this out first.